### PR TITLE
Adding Metrics endpoint to cfn templates

### LINKED ISSUE #88: Create Production deployment AWS Cloudformation template for TorchServe
For Production deployment of TorchServe, provide AWS Cloudformation templates that support following configurations:

- [x] Simple EC2 deployment
- [ ] Sagemaker deployment with autoscaling
- [ ] K8s/EKS deployment with autoscaling 

Each of the production deployment templates should include configuration with:

- TorchServe inference node(s)
- TorchServe Model management node
- NGINX loadbalancer
- Metrics collection node - Statsd/Prometheus/CloudWatch
- Visualization Dashboard for metrics - Cloudwatch dashboard / Grafana
- Log aggregation on CloudWatch/Fluentd (or similar) 
- Secure deployment option using HTTPS 
- Option to select CPU or GPU deployment for the Inference nodes (this could be a parameter to select AWS instance type for the Inference nodes).

### PR DESCRIPTION
**Pull request by maaquib**
## Description

Adding Metrics endpoint to cfn templates

Fixes #88

## Type of change

Please delete options that are not relevant.
- [X] New feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update

## Feature/Issue validation/testing

Verified by creating stacks using commands in the README

### CODE REVIEW

File: cloudformation/ec2-asg.yaml [hunk 1]
```diff
@@ -349,6 +365,19 @@ Resources:
       Protocol: HTTP
       UnhealthyThresholdCount: 5
       VpcId: !Ref 'VPC'
+  ALBMetricsTargetGroup:
+    Type: AWS::ElasticLoadBalancingV2::TargetGroup
+    Properties:
+      HealthCheckIntervalSeconds: 30
+      HealthCheckTimeoutSeconds: 5
+      HealthyThresholdCount: 3
+      HealthCheckPath: /ping
```
Comment (chauhang) [inline cloudformation/ec2-asg.yaml:374]: @maaquib Shouldn't this be a health check for the metric endpoint vs using the same /ping for all the endpoints?

**Review by chauhang — CHANGES_REQUESTED** [review]
Please see comments inline

File: cloudformation/ec2-asg.yaml (hunk 1 above)
Comment (maaquib) [reply cloudformation/ec2-asg.yaml:374]: @chauhang We only have a single health check endpoint right now. I'll create an issue to have health checks for all the APIs

Comment (chauhang) [reply cloudformation/ec2-asg.yaml:374]: @maaquib Created #735

**Review by chauhang — APPROVED** [review]

**Review by dhanainme — APPROVED** [review]

**Review by dhanainme — APPROVED** [review]

**Pull request merged**